### PR TITLE
Feature/typing

### PR DESCRIPTION
**Pull request by ghbm-itk**
### Fixed

- Refactoring for more correct typing.
- Get-functions in db_util now works in SQLITE.

### CODE REVIEW

File: OpenOrchestrator/common/crypto_util.py [hunk 1]
```diff
@@ -4,7 +4,7 @@
 from cryptography.exceptions import InvalidSignature
 
 
-_encryption_key: str = None
+_encryption_key: str | None = None
```
Comment (bakspace-itk) [inline OpenOrchestrator/common/crypto_util.py:7]: So this is to allow the value of None, or make the code more explicit?

File: OpenOrchestrator/common/crypto_util.py [hunk 2]
```diff
@@ -43,10 +43,16 @@ def encrypt_string(data: str) -> str:
...
         str: The encrypted string.
+
+    Raises:
+        RuntimeError: If the encryption key has not been set.
     """
-    data = data.encode()
-    data = Fernet(_encryption_key).encrypt(data)
-    return data.decode()
+    if not _encryption_key:
+        raise RuntimeError("Can't encrypt without an encryption key.")
```
Comment (bakspace-itk) [inline OpenOrchestrator/common/crypto_util.py:51]: Explicit errors rather than relying on generics is always nice

File: OpenOrchestrator/database/db_util.py [hunk 3]
```diff
@@ -1,9 +1,10 @@
 """This module handles the connection to the database in OpenOrchestrator."""
 
 from datetime import datetime
-from typing import Callable, TypeVar, ParamSpec
+from typing import TypeVar, ParamSpec
+from uuid import UUID
 
-from croniter import croniter
+from croniter import croniter  # type: ignore
```
Comment (bakspace-itk) [inline OpenOrchestrator/database/db_util.py:7]: Not sure I understand, is this for the linter?

File: OpenOrchestrator/database/db_util.py [hunk 4]
```diff
@@ -48,19 +49,25 @@ def connect(conn_string: str) -> bool:
...
+        _connection_engine.dispose()
     _connection_engine = None
 
 
-def catch_db_error(func: Callable[P, T]) -> Callable[P, T]:
-    """A decorator that catches errors in SQL calls."""
-    def inner(*args, **kwargs) -> T:
-        if _connection_engine is None:
-            raise RuntimeError("Not connected to Database")
+def _get_session() -> Session:
```
Comment (bakspace-itk) [inline OpenOrchestrator/database/db_util.py:57]: I like that the error handling is in the function rather than having a function for catching errors

File: OpenOrchestrator/database/db_util.py [hunk 5]
```diff
@@ -69,126 +76,132 @@ def get_conn_string() -> str:
...
+
+    if not trigger:
+        raise ValueError(f"No trigger with the given id: {trigger_id}")
+
+    return trigger
 
 
-@catch_db_error
-def get_all_triggers() -> tuple[Trigger]:
+def get_all_triggers() -> tuple[Trigger, ...]:
```
Comment (bakspace-itk) [inline OpenOrchestrator/database/db_util.py:125]: the tuple[Trigger, ...] thing is a bit weird, but I think I get it. You can return a tuble with more than just Triggers in it? What would that be, errors?

File: OpenOrchestrator/database/db_util.py [hunk 6]
```diff
@@ -647,9 +659,15 @@ def begin_queue_trigger(trigger_id: str) -> None:
     Returns:
         bool: True if the trigger was 'idle' and now 'running'.
     """
-    with Session(_connection_engine) as session:
+    if isinstance(trigger_id, str):
+        trigger_id = UUID(trigger_id)
```
Comment (bakspace-itk) [inline OpenOrchestrator/database/db_util.py:663]: Nice and simple solution

File: OpenOrchestrator/orchestrator/datetime_input.py [hunk 7]
```diff
@@ -19,7 +19,7 @@ def __init__(self, label: str, on_change: Optional[Callable[..., Any]] = None, a
             on_change: A callable to execute on change. Defaults to None.
             allow_empty: Whether to allow an empty input on validation. Defaults to False.
         """
-        super().__init__(label, value=None, on_change=on_change)
+        super().__init__(label, value='', on_change=on_change)
```
Comment (bakspace-itk) [inline OpenOrchestrator/orchestrator/datetime_input.py:22]: How is this better?

File: OpenOrchestrator/scheduler/connection_frame.py [hunk 8]
```diff
@@ -91,6 +91,6 @@ def new_key(self):
         """Creates a new encryption key and inserts it
         into the key entry.
         """
-        key = crypto_util.generate_key()
+        key = crypto_util.generate_key().decode()
```
Comment (bakspace-itk) [inline OpenOrchestrator/scheduler/connection_frame.py:94]: Why this change?

File: OpenOrchestrator/database/db_util.py (hunk 5 above)
Comment (ghbm-itk) [reply OpenOrchestrator/database/db_util.py:125]: tuple[Trigger] means "A tuple of ONE Trigger".
tuple[Trigger, ...] means "A tuple of any number of Triggers"

File: OpenOrchestrator/database/db_util.py (hunk 3 above)
Comment (ghbm-itk) [reply OpenOrchestrator/database/db_util.py:7]: Yes

**Review by bakspace-itk — APPROVED** [review]
